beacon-chain/blockchain/forkchoice/process_block.go

```diff
@@ -515,3 +520,31 @@ func (s *Store) saveInitState(ctx context.Context, state *pb.BeaconState) error
 	}
 	return nil
 }
+
+// This filters block roots that are not known as head root and finalized root in DB.
+// It serves as the last line of defence before we prune states.
+func (s *Store) filterBlockRoots(ctx context.Context, roots [][32]byte) ([][32]byte, error) {
```

Comment:
This function name is a little vague, do you think you can make it more descriptive? Maybe `filterPrunableStates`?

Reply:
I think it's fine. The function is filtering block roots not state and the comment explicitly specified what kind of roots are getting filtered